packages/datetime/src/dateInput.tsx

```diff
@@ -155,13 +164,21 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
         const dateString = this.state.isInputFocused ? this.state.valueString : this.getDateString(this.state.value);
         const date = this.state.isInputFocused ? moment(this.state.valueString, this.props.format) : this.state.value;
 
-        const popoverContent = (
+        const sharedProps: IDatePickerBaseProps = {
+            ...this.props,
+            canClearSelection: this.props.canClearSelection,
+            defaultValue: null,
```

Comment:
definitely don't need `defaultValue` (uncontrolled) since you're setting `value` (controlled)